actors/miner/src/deadline_state.rs

```diff
@@ -89,9 +83,9 @@ impl Deadlines {
         store: &BS,
         deadline_idx: u64,
         deadline: &Deadline,
-    ) -> anyhow::Result<()> {
+    ) -> Result<(), ActorError> {
         if deadline_idx >= policy.wpost_period_deadlines {
-            return Err(anyhow!("invalid deadline {}", deadline_idx));
+            return Err(actor_error!(illegal_argument, "invalid deadline {}", deadline_idx));
```

Comment:
This is an interesting case that's illegal state in Go (by way of no exit code being specified at this level), but using illegal argument would be better.